types/types.go

```diff
@@ -5,6 +5,46 @@ import (
 	"time"
 )
 
+// ImageTransport is a top-level namespace for ways to refer to an image.
+// It should generally correspond to ImageSource/ImageDestination implementations.
+//
+// Note that ImageTransport is based on "ways the users refer to images", not necessarily on the underlying physical transport.
+// For example, all Docker References would be used within a single "docker" transport, regardless of whether the images are pulled over HTTP or HTTPS
+// (or, even, IPv4 or IPv6).
+//
+// OTOH all images using the same transport should (apart from format versions), be interoperable.
+// For example, several different ImageTransport implementations may be based on local filesystem paths,
+// but using completely different formats for the contents of that path (a single tar file, a directory containing tarballs, a fully expanded container filesystem, ...)
+// See also containers/image.KnownTransports.
+type ImageTransport interface {
+	// Name returns the name of the transport, which must be unique among other transports.
+	Name() string
+	// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an ImageReference.
+	ParseReference(reference string) (ImageReference, error)
+}
+
+// ImageReference is an abstracted way to refer to an image or a namespace containing images, namespaced within an ImageTransport
+type ImageReference interface {
```

Comment:
well, we already stated somewhere is by any mean to be used other than a simple debug functionality so I'm fine with this. We probably need to make `dir:` hidden under `--debug` flag so it can only be used if running in debug (no need to think about this now)